File: API/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Product
    {
        [Key]
        public int ProductId { get; set; }

        public string? Name { get; set; }

        public long Price { get; set; }

        public string? Description { get; set; }

        public string? ImageUrl { get; set; }


        public ProductType? ProductType { get; set; }
        public int ProductTypeId { get; set; }


        public Brand? Brand { get; set; }
        public int BrandId { get; set; }

        public List<ProductSize> ProductSizes { get; set; } = new List<ProductSize>();

         public void AddItem(Size size, int quantity)
        {
            if (ProductSizes.All(prSize => prSize.ProductId!=this.ProductId && prSize.SizeId != size.Id))
            {
                ProductSizes.Add(new ProductSize { ProductId = this.ProductId, SizeId = size.Id, QuantityInStock = quantity });
                return;
            }

            var existingItem = ProductSizes.FirstOrDefault(prSize => prSize.ProductId == this.ProductId && prSize.SizeId == size.Id);
            if (existingItem != null) existingItem.QuantityInStock += quantity;
        }

    }
}

File: client/src/app/models/product.ts


export interface Product {
    productId: number
    name: string
    price: number
    description: string
    imageUrl: string
    productType: ProductType
    productTypeId: number
    brand: Brand
    brandId: number
    productSizes?: ProductSize[]

  }
  
  export interface ProductType {
    productTypeId: number
    name: string
  }


  export interface Brand {
    brandId: number
    name: string
  }
  
  export interface ProductSize {
    id: number
    size: Size
    sizeId: number
    product: string
    productId: number
    quantityInStock: number
  }
  
  export interface Size {
    id: number
    sizeOfProduct: string
    productSizes: string[]
  }

  export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    types: string[];
    brands: Brand[];
    pageNumber: number;
    pageSize: number;
    productSizes: string[];
}

export interface Size {
  id: number
  sizeOfProduct: string
  productSizes: string[]
}